Whiskey/Functions/Invoke-WhiskeyTask.ps1

```diff
@@ -178,105 +178,119 @@ function Invoke-WhiskeyTask
         }
     }
 
-    $workingDirectory = $TaskContext.BuildRoot
-    if( $Parameter['WorkingDirectory'] )
-    {
-        $workingDirectory = $Parameter['WorkingDirectory'] | Resolve-WhiskeyTaskPath -TaskContext $TaskContext -PropertyName 'WorkingDirectory'
-    }
-
-    $taskTempDirectory = ''
-    $requiredTools = Get-RequiredTool -CommandName $task.CommandName
-    $startedAt = Get-Date
-    $result = 'FAILED'
+    # Start every task in the BuildRoot.
+    Push-Location $TaskContext.BuildRoot
     $currentDirectory = [IO.Directory]::GetCurrentDirectory()

```

Comment:
`$currentDirectory` becomes 'non current' soon, as opposed to eg. `$originalDirectory`. It makes the restore easier to pair later.